File: sgi-frontend/src/types/importaciones.ts
export interface Importacion {
    id?: number;
    ruc?: string;
    razon_social?: string;
    sector?: string;
    score?: number;
    agentes_distintos?: number;
    total_embarques?: number;
    meses_activos?: number;
    fob_promedio?: number;
    via_predominante?: string;
    paises_principales?: string;
    ultima_importacion?: string;
    dias_desde_ultima?: number;
}

export interface ImportacionResponse {
    data: Importacion[];
    total: number;
    page: number;
    page_size: number;
}

File: sgi-backend/app/schemas/comercial/importaciones.py
from pydantic import BaseModel, ConfigDict
from typing import Optional
from decimal import Decimal

class ImportacionBase(BaseModel):
    id: int
    ruc: Optional[str] = None
    razon_social: Optional[str] = None
    sector: Optional[str] = None
    score: Optional[Decimal] = None
    agentes_distintos: Optional[int] = None
    total_embarques: Optional[int] = None
    meses_activos: Optional[int] = None
    fob_promedio: Optional[Decimal] = None
    via_predominante: Optional[str] = None
    paises_principales: Optional[str] = None
    ultima_importacion: Optional[str] = None
    dias_desde_ultima: Optional[int] = None

class ImportacionResponse(ImportacionBase):
    model_config = ConfigDict(from_attributes=True)

class ImportacionPagination(BaseModel):
    total: int
    page: int
    page_size: int
    data: list[ImportacionResponse]